ccm_web/server/src/lti/lti.service.ts

```diff
@@ -94,16 +94,26 @@ export class LTIService implements BeforeApplicationShutdown {
         logger.error(`Something went wrong while creating user with loginId ${loginId}: ${logMessageEnding}`)
         return createLaunchErrorResponse(res)
       }
-      // More data will be added to the session here later
-      const course = {
-        id: courseId,
-        roles: (roles.length > 0) ? roles.split(',') : [] // role won't be empty but adding a validation for safety
-      }
-      const sessionData = {
-        course: course,
-        isRootAdmin: isRootAdmin
+    
+      try {
+        // More data will be added to the session here later
+        const course = {
+          id: Number(courseId),
+          roles: (roles.length > 0) ? roles.split(',') : [] // role won't be empty but adding a validation for safety
+        }
+        const sessionData = {
+          course: course,
+          isRootAdmin: isRootAdmin
+        }
+        req.session.data = sessionData
+      } catch (e) {
+        const logMessageEnding = e instanceof Error
+          ? `error ${String(e.name)} due to ${String(e.message)}`
+          : String(e)
+        logger.error(`Failed to build session data with course ID ${courseId} and roles ${roles}: ${logMessageEnding}`)
+        return createLaunchErrorResponse(res)
```

Comment:
Why did you feel the need to adding a try-catch?

Reply:
I was following a previous example of us handling errors from course ID changing to integer that I think you linked before: https://github.com/tl-its-umich-edu/canvas-app-explorer/commit/e6fc3ac1167f86e7574ded62304fdb406cd1aee3

Since we're reformatting roles and courseId for the sessionData , I felt it made sense to have a descriptive log if the formatting fails

Reply:
ok